### ISSUE TITLE
get_levels() cache needs flush

### ISSUE THREAD

**Issue by tannerm** [restrictcontentpro/restrict-content-pro#646]
I don't have time to look into this one atm, but the cache for RCP_Levels->get_levels() does not appear to be flushed when a new level is created.

**Comment by mindctrl** [comment]
I think we should move to the Transients API. See this comment: https://github.com/restrictcontentpro/restrict-content-pro/issues/790#issuecomment-269360954

**Comment by pippinsplugins** [comment]
:+1:

**Comment by pippinsplugins** [comment]
Going to move this to 2.7.5 as it's not super important.

**Comment by nosegraze** [comment]
I've been getting a lot of tickets about this, particularly for users on the WordPress.com Business plan.

**Comment by pippinsplugins** [comment]
@nosegraze could you go ahead and work up a PR and we can include this in the next point release?

**Comment by nosegraze** [comment]
PR: #1671

For now this PR just fixes the problem with the WP object cache not being cleared. I think the main problem was the cache key included the `$where` clause, but whenever we did `wp_cache_delete()` we didn't include that in the key name. I just removed the `$where` clause from the key name when we set it, as I don't see it as necessary anyway.

I created a new issue for moving everything to the transients API as that'll be a bigger change: https://github.com/restrictcontentpro/restrict-content-pro/issues/1670
